File: backend.spring/src/main/scala/com/skewwhiffy/auraltester/notes/Octave.scala
package com.skewwhiffy.auraltester.notes

object Octave {
  lazy val default: Octave = new Octave(0)
}

class Octave(val offsetFromDefault: Int) {
  lazy val up: Octave = new Octave(offsetFromDefault + 1)

  lazy val down: Octave = new Octave(offsetFromDefault - 1)

  def >(other: Octave): Boolean = offsetFromDefault > other.offsetFromDefault

  def <(other: Octave): Boolean = offsetFromDefault < other.offsetFromDefault

  def >=(other: Octave): Boolean = offsetFromDefault > other.offsetFromDefault || this == other

  def <=(other: Octave): Boolean = offsetFromDefault < other.offsetFromDefault || this == other

  override def equals(obj: Any): Boolean = obj match {
    case it: Octave => it.offsetFromDefault == offsetFromDefault
    case _ => false
  }
}

File: backend/src/main/kotlin/com/skewwhiffy/notation/model/Octave.kt
package com.skewwhiffy.notation.model

data class Octave(val offsetFromDefault: Int) {
  companion object {
    val default: Octave = Octave(0)
  }

  val up: Octave get() = Octave(offsetFromDefault + 1)

  val down: Octave get() = Octave(offsetFromDefault - 1)
  /*

  def >(other: Octave): Boolean = offsetFromDefault > other.offsetFromDefault

  def <(other: Octave): Boolean = offsetFromDefault < other.offsetFromDefault

  def >=(other: Octave): Boolean = offsetFromDefault > other.offsetFromDefault || this == other

  def <=(other: Octave): Boolean = offsetFromDefault < other.offsetFromDefault || this == other
}
   */
}